src/containerapp/azext_containerapp/tests/latest/test_containerapp_gpu.py

```diff
@@ -0,0 +1,86 @@
+
+        while containerapp_env["properties"]["provisioningState"].lower() == "waiting":
+            time.sleep(5)
+            containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env)).get_output_in_json()
+        app1 = self.create_random_name(prefix='app1', length=24)
+        self.cmd(f'containerapp create -n {app1} -g {resource_group} --image mcr.microsoft.com/azuredocs/samples-tf-mnist-demo:gpu --environment {env} -w {gpu_default_name} --min-replicas 1 --cpu 0.1 --memory 0.1', checks=[
+            JMESPathCheck("properties.provisioningState", "Succeeded"),
+            JMESPathCheck("properties.workloadProfileName", gpu_default_name),
+            JMESPathCheck('properties.template.containers[0].resources.cpu', '0.1'),
+            JMESPathCheck('properties.template.containers[0].resources.memory', '0.1Gi'),
+            JMESPathCheck('properties.template.containers[0].resources.gpu', '1'),
+            JMESPathCheck('properties.template.scale.minReplicas', '1'),
+            JMESPathCheck('properties.template.scale.maxReplicas', '10')
+        ])
+
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer(location="eastus2")
+    def test_containerapp_create_enable_consumption_gpu(self, resource_group):
+        self.cmd('configure --defaults location={}'.format("northeurope"))
+        env = self.create_random_name(prefix='consumption-gpu-env', length=24)
+        self.cmd('containerapp env create -g {} -n {} --logs-destination none --enable-workload-profiles'.format(
+            resource_group, env), expect_failure=False, checks=[
+            JMESPathCheck("name", env),
+            JMESPathCheck("properties.provisioningState", "Succeeded"),
+            JMESPathCheck("length(properties.workloadProfiles)", 1),
+            JMESPathCheck('properties.workloadProfiles[0].name', "Consumption", case_sensitive=False),
+            JMESPathCheck('properties.workloadProfiles[0].workloadProfileType', "Consumption", case_sensitive=False),
+        ])
+        consumption_gpu_wp_name = "Consumption-T4"
+
+        self.cmd("az containerapp env workload-profile set -g {} -n {} --workload-profile-name {consumption_gpu_wp_name} --workload-profile-type Consumption_GPU_NC8as_T4".format(
+            resource_group, env), expect_failure=False)
+
+        containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env)).get_output_in_json()
+
+        while containerapp_env["properties"]["provisioningState"].lower() == "waiting":
+            time.sleep(5)
+            containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env)).get_output_in_json()
+        app1 = self.create_random_name(prefix='app1', length=24)
+        self.cmd(f'containerapp create -n {app1} -g {resource_group} --image mcr.microsoft.com/azuredocs/samples-tf-mnist-demo:gpu --environment {env} -w {consumption_gpu_wp_name} --cpu 0.1 --memory 0.1 --gpu 1', checks=[
```

Comment:
please add test to cover all commands that support --gpu (`az containerapp create/update`, `az containerapp compose create`.